samples/wifi/radio_test/src/nrf_wifi_radio_ficr_shell.c

```diff
@@ -0,0 +1,346 @@
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr_util.h>
+#include <zephyr_fmac_main.h>
+#include <nrf_wifi_radio_test_shell.h>
+#include "ficr_prog.h"
+
+LOG_MODULE_REGISTER(otp_prog, CONFIG_WIFI_LOG_LEVEL);
+
+
+static void disp_location_status(char *region, uint32_t ret)
+{
+
+	switch (ret) {
+
+	case 0x00000000:
+		LOG_INF("%s Region is locked!!!\n", region);
+		break;
+	case 0x50FA50FA:
+		LOG_INF("%s Region is open for R/W\n", region);
+		break;
+	case 0xFFFFFFFF:
+		LOG_INF("%s Region is unprogrammed - program to enable R/W\n", region);
+		break;
+	default:
+		LOG_INF("%s Region is in invalid state\n", region);
+		break;
+	}
+
+	LOG_INF("\n");
+}
+
+static void disp_fields_status(uint32_t flags)
+{
+	if (flags & (~QSPI_KEY_FLAG_MASK))
+		LOG_INF("QSPI Keys are not programmed in OTP\n");
+	else
+		LOG_INF("QSPI Keys are programmed in OTP\n");
+
+	if (flags & (~MAC0_ADDR_FLAG_MASK))
+		LOG_INF("MAC0 Address are not programmed in OTP\n");
+	else
+		LOG_INF("MAC0 Address is programmed in OTP\n");
+
+	if (flags & (~MAC1_ADDR_FLAG_MASK))
+		LOG_INF("MAC1 Address are not programmed in OTP\n");
+	else
+		LOG_INF("MAC1 Address is programmed in OTP\n");
+
+	if (flags & (~CALIB_XO_FLAG_MASK))
+		LOG_INF("CALIB_XO is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_XO is programmed in OTP\n");
+
+	if (flags & (~CALIB_PDADJMCS7_FLAG_MASK))
+		LOG_INF("CALIB_PDADJMCS7 not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_PDADJMCS7 is programmed in OTP\n");
+
+	if (flags & (~CALIB_PDADJMCS0_FLAG_MASK))
+		LOG_INF("CALIB_PDADJMCS0 is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_PDADJMCS0 is programmed in OTP\n");
+
+	if (flags & (~CALIB_MAXPOW2G4_FLAG_MASK))
+		LOG_INF("CALIB_MAXPOW2G4 is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_MAXPOW2G4 is programmed in OTP\n");
+
+	if (flags & (~CALIB_MAXPOW5G0MCS7_FLAG_MASK))
+		LOG_INF("CALIB_MAXPOW5G0MCS7 is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_MAXPOW5G0MCS7 is programmed in OTP\n");
+
+	if (flags & (~CALIB_MAXPOW5G0MCS0_FLAG_MASK))
+		LOG_INF("CALIB_MAXPOW5G0MCS0 is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_MAXPOW5G0MCS0 is programmed in OTP\n");
+
+	if (flags & (~CALIB_RXGAINOFFSET_FLAG_MASK))
+		LOG_INF("CALIB_RXGAINOFFSET is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_RXGAINOFFSET is programmed in OTP\n");
+
+	if (flags & (~CALIB_TXPOWBACKOFFT_FLAG_MASK))
+		LOG_INF("CALIB_TXPOWBACKOFFT is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_TXPOWBACKOFFT is programmed in OTP\n");
+
+	if (flags & (~CALIB_TXPOWBACKOFFV_FLAG_MASK))
+		LOG_INF("CALIB_TXPOWBACKOFFV is not programmed in OTP\n");
+	else
+		LOG_INF("CALIB_TXPOWBACKOFFV is not programmed in OTP\n");
+}
+
+
+static int nrf_wifi_radio_test_otp_get_status(const struct shell *shell,
+					size_t argc,
+					char  *argv[])
+{
+	uint32_t ret, err;
+	uint32_t val[OTP_MAX_WORD_LEN];
+
+	/* read all the OTP memory */
+	err = read_otp_memory(0, &val[0], OTP_MAX_WORD_LEN);
```

Comment:
This could be made as `err = read_otp_memory(0, val, sizeof(val));` so that it will work even if size of val changes in the future.

Reply:
will look into this for the next revision